tests/test_runner/test_hooks.py

```diff
@@ -39,6 +39,85 @@
 sys.modules['petrel_client.client'] = MagicMock()
 
 
+def test_optimizerhook():
+
+    class Model(nn.Module):
```

Comment:
The unit test only test the case when conv3 is not used in the forwarwd pass. We should also test the case when conv3 is forwarded but not used for computing loss.

Reply:
This case has been tested when the loss is `x1.sum()` then conv2 would be detected. Line 98